src/Simulation/Common/IQuantumProcessor.cs

```diff
@@ -0,0 +1,587 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Simulation.Common
+{
+    /// <summary>
+    /// An interface for implementing QDK target quantum machines that work on a quantum circuit level. 
+    /// </summary>
+    /// <remarks>
+    /// To implement a target machine that executes quantum commands, implement this interface.
+    /// Consider using <see cref="QuantumProcessorBase"/> as a stub for easy impementation of this interface.
+    /// Implementors of <see cref="IQuantumProcessor"/> interface do not manage qubits on their own.
+    /// All qubit management (allocation, dealocation, etc.) is done by the caller of this interface.
+    /// Implementors are notified when qubits are allocated, released, borrowed and returned allowing them to react to these events if necessary.
+    /// </remarks>
+    public interface IQuantumProcessor
+    {
+        /// <summary>
+        /// Called when <a href="https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.intrinsic.x">Microsoft.Quantum.Intrinsic.X</a> is called in Q#.
+        /// In Q# the operation applies X gate to <paramref name="qubit"/>. The gate is given by matrix X=((0,1),(1,0)).
```

Comment:
```suggestion
        /// When this is invoked, it is expected that the X gate gets applied to the state of the given <paramref name="qubit"/>. The gate is given by matrix X=((0,1),(1,0)).
```
I suggest all descriptions are modified from `In Q# the operation applies...` to a text similar to what I suggest above.